examples/wasm-cc/Dockerfile-proxy

```diff
@@ -1,4 +1,4 @@
-FROM envoyproxy/envoy:dev
+FROM envoyproxy/envoy:dev@sha256:6fbf6036f7db3e0b557fd7da31f63c5f733c3b4cf0d477fab4b6647b1855fca0
```

Comment:
these cant be pinned unfortunately - not without some more complex scheme

in CI the images are downloaded and loaded into the Docker daemon - so it uses the CI version not the hub one

we could potentially use a build arg and a default - but tbh my reading of the ossf situation is that probably wont help, and would add a lot of complexity, even if a bit more secure for some cases